### PR TITLE
picom: misc fixes

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

- Build option `build_docs` was renamed to `with_docs` quite sometime ago: yshui/picom@3f2a671. This means that the manpage were not being build.
- Remove CFLAG `-fno-strict-aliasing` that seems to not be used anywhere in the upstream repository. This was probably added for `compton` a long time ago and never removed, since I can't find the commit that added this.
- Use release build. This is recommended upstream: https://github.com/yshui/picom#to-build.

Also, add myself as maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
~
❯ nix path-info -S /nix/store/zm67sy1dgccz8iz7gy9vw0s01cj5i2r7-picom-8.2/bin/picom # before
/nix/store/zm67sy1dgccz8iz7gy9vw0s01cj5i2r7-picom-8.2	 156816464

~
❯ nix path-info -S /nix/store/s1m8hars4bzjsidlkkphzycaaq24a46z-picom-8.2/bin/picom # after
/nix/store/s1m8hars4bzjsidlkkphzycaaq24a46z-picom-8.2	 156825808
```

### CODE REVIEW

**Comment by thiagokokada** [comment]
This will probably conflict with this PR: https://github.com/NixOS/nixpkgs/pull/85490

File: pkgs/applications/window-managers/picom/default.nix [hunk 1]
```diff
@@ -35,10 +35,10 @@ stdenv.mkDerivation rec {
     libxdg_basedir
   ];
 
-  NIX_CFLAGS_COMPILE = "-fno-strict-aliasing";
+  mesonBuildType = "release";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/picom/default.nix:38]: I wonder why this does not default to release.

Comment (SuperSandro2000) [reply pkgs/applications/window-managers/picom/default.nix:38]: I mean the option in nixpkgs. I think the default value is plain.

Comment (thiagokokada) [reply pkgs/applications/window-managers/picom/default.nix:38]: FYI, diff in derivation size is quite small. I failed at copy and paste in OP :facepalm: .

Comment (thiagokokada) [reply pkgs/applications/window-managers/picom/default.nix:38]: The binary size itself decreased from 396k->394k (but the derivation size is slightly larger thanks to the manpages probably).

**Comment by thiagokokada** [comment]
Fixed conflict with PR https://github.com/NixOS/nixpkgs/pull/108555

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108530` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>picom</li>
</ul>
</details>

**Pull request merged**